### PR TITLE
Use setup.cfg with adding python_requires

### PR DESCRIPTION
**Pull request by chezou**
This PR to use setup.cfg instead of setup.py also adding `python_requires`, describing explicit supported version for Python

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,3 +1,52 @@
+[metadata]
+name = pytd
+version = 1.4.0
```
Comment (chezou) [inline setup.cfg:3]: Unfortunately `version = attr: pytd.__version__` doesn't work well while there was improvement for it
https://stackoverflow.com/questions/58202909/modulenotfounderror-when-using-setup-cfg-and-version-accessed-with-attr
https://github.com/pypa/setuptools/pull/1753

**Comment by chezou** [comment]
Hmm, we hit  `TypeError: Cannot interpret '<attribute 'dtype' of 'numpy.generic' objects>' as a data type` error. Maybe numpy 1.2.0 release affects it.

File: doc/contributing.rst [hunk 2]
```diff
@@ -72,11 +72,11 @@ The ``doc/`` folder is monitored and automatically published by `Read the Docs <
 Releasing
 ---------
 
-Update version in ``pytd/version.py``. Set it to ``1.0.0``, for example:
+Update version in ``setup.cfg``. Set it to ``1.0.0``, for example:
 
 .. code:: py
 
-   __version__ = "1.0.0"
+   version = 1.0.0
```
Comment (takuti) [inline doc/contributing.rst:79]: Let's make it more interpretable as `setup.cfg`

```
.. code:: ini

   [metadata]
   name = pytd
   version = 1.0.0
```

File: setup.py [hunk 3]
```diff
@@ -7,75 +7,6 @@
 a shorter time. It eventually makes your day-to-day data analytics work more
 efficient and productive.
 """
```
Comment (takuti) [inline setup.py:9]: Since the config file no longer refers `__doc__`, we can delete the top-level statements.

**Review by takuti — APPROVED** [review]

**Comment by chezou** [comment]
Thanks, applied feedbacks on 865e9df

**Pull request merged**